examples/src/java/org/pantsbuild/example/3rdparty_jvm.md

```diff
@@ -291,3 +291,237 @@ time pants is run (otherwise, pants will cache it):
+that target.
+
+#### Invalidation
+
+This is a change from the current behavior where all downstream targets will
+have their compile invalidated.
+
+With strict_deps, a change to a library target will only invalidate the
+compile artifacts of dependent targets if they have
+
+1. A direct dependency on the changed library target, or
+2. A dependency on it via the export graph.
+
+Before, if a target depending on a library had an implicit dependency on a
+transitive dependency of the library target it would just work. The transitive
+dependency would end up on the dependent target’s compile classpath and that
+target would compile happily. Strict deps removes that implicit transitivity,
+which exposes undeclared dependencies.
+
+This introduces two new sources of errors.
+
+1. When adding a new library dependency to a target, if the library dependency
+requires some of its dependencies to be on the compile classpath, the compile
+can fail if those dependencies don’t make it on the classpath. These errors
+are a bit weird and will be confusing at first.
+2. When a library target adds a new publicly visible dependency, targets that
+depend on the library target will fail to compile with the same kind of error
+message as 1.
+
+### Errors, Exports, and Missing-deps-suggest
+
+The errors sometimes have different messages, but usually in Scala they look
+something like this:
+
+    :::bash
+    [error] <path-to-current-sources>/<Some-file>.scala:71:32: Symbol 'type
+            <Some-type>' is missing from the classpath.
+    [error] This symbol is required by '<Some-other-type>.timeout'.
+    [error] Make sure that type <Some-type>is in your classpath and check for
+            conflicting dependencies with `-Ylog-classpath`.
+    [error] A full rebuild may help if <Some-other-type>.class was compiled
+            against an incompatible version of <none>.<Some-type>.
+    [error]         Some-file.code
+    [error]
+
+**NOTE: It's likely the source snippet may not be related to the error**
+
+There are two possible solutions for this error. One is to identify the
+dependency that is missing and add it to your target directly.
+
+#### Missing-deps-suggest
+
+Missing-deps-suggest is a feature that will try to identify the target that
+failed to compile and suggest a dependency that may be missing.
+
+If this doesn't provide a solution, then the second solution to this error
+is exporting the missing type's target in the dependency you depend on.
+
+#### Exports
+
+A target should export any dependencies which provide types or symbols that
+are exposed in its public API. For instance, if target A has a method returning
+something of type X, it should export the target which defines the type X.
+
+Exports reintroduce limited transitivity to the dependencies provided at
+compile time. When a target exports one of its dependencies, that dependency
+will be on the classpath of any target that depends on it.
```

Comment:
Maybe these would be better?
> What are exports? Exported targets are targets that are necessary to compile against the target doing the exporting. For example, javac needs to know about the super classes of a class in order for another class to inherit from it. If the API of a class inside a target expects subclassing, then the targets that contain that class's superclasses would need to be exported.

> Exports are transitive. This means that if a target exports one of its dependencies and that exported dependency has exports, those exports will also be end up on the classpath. This allows the export graph to model requirements created by type hierarchies.

Reply:
The first paragraph you are suggesting is a little convoluted to me. I think a concrete example is more clear.

The second paragraph makes sense, I'll definitely include that.